src/modules/cpu/host_tensor_augmentations.hpp

```diff
@@ -2825,6 +2825,1420 @@ RppStatus blend_i8_i8_host_tensor(Rpp8s *srcPtr1,
+                    srcPtrTempG += vectorIncrementPerChannel;
+                    srcPtrTempB += vectorIncrementPerChannel;
+                    dstPtrTemp += vectorIncrement;
+                }
+                for (; vectorLoopCount < bufferLength; vectorLoopCount++)
+                {
+                    RpptFloatRGB pixel;
+                    pixel.R = (Rpp32f) *srcPtrTempR;
+                    pixel.G = (Rpp32f) *srcPtrTempG;
+                    pixel.B = (Rpp32f) *srcPtrTempB;
+                    compute_color_twist_host(&pixel, brightnessParam, contrastParam, hueParam, saturationParam);
+                    dstPtrTemp[0] = (Rpp16f) RPPPIXELCHECKF32(pixel.R);
+                    dstPtrTemp[1] = (Rpp16f) RPPPIXELCHECKF32(pixel.G);
+                    dstPtrTemp[2] = (Rpp16f) RPPPIXELCHECKF32(pixel.B);
+
+                    srcPtrTempR++;
+                    srcPtrTempG++;
+                    srcPtrTempB++;
+                    dstPtrTemp += 3;
+                }
+
+                srcPtrRowR += srcDescPtr->strides.hStride;
+                srcPtrRowG += srcDescPtr->strides.hStride;
+                srcPtrRowB += srcDescPtr->strides.hStride;
+                dstPtrRow += dstDescPtr->strides.hStride;
+            }
+        }
+
+        // Color Twist without fused output-layout toggle (NHWC -> NHWC)
+        else if ((srcDescPtr->c == 3) && (srcDescPtr->layout == RpptLayout::NHWC) && (dstDescPtr->layout == RpptLayout::NHWC))
+        {
+            Rpp16f *srcPtrRow, *dstPtrRow;
+            srcPtrRow = srcPtrChannel;
+            dstPtrRow = dstPtrChannel;
+
+            for(int i = 0; i < roiPtr->xywhROI.roiHeight; i++)
+            {
+                Rpp16f *srcPtrTemp, *dstPtrTemp;
+                srcPtrTemp = srcPtrRow;
+                dstPtrTemp = dstPtrRow;
+
+                int vectorLoopCount = 0;
+                for (; vectorLoopCount < alignedLength; vectorLoopCount += vectorIncrement)
+                {
+                    Rpp32f srcPtrTemp_ps[24];
+                    Rpp32f dstPtrTemp_ps[24];
+                    for(int cnt = 0; cnt < vectorIncrement; cnt++)
+                        srcPtrTemp_ps[cnt] = (Rpp32f) srcPtrTemp[cnt];
+#if __AVX2__
+                    __m256 p[3];
+                    rpp_simd_load(rpp_load24_f32pkd3_to_f32pln3_avx, srcPtrTemp_ps, p);    // simd loads
+                    compute_color_twist_24_host(p[0], p[1], p[2], pColorTwistParams);    // color_twist adjustment
+                    rpp_simd_store(rpp_store24_f32pln3_to_f32pkd3_avx, dstPtrTemp_ps, p);    // simd stores
+#else
+                    __m128 p[4];
+                    rpp_simd_load(rpp_load12_f32pkd3_to_f32pln3, srcPtrTemp_ps, p);    // simd loads
+                    compute_color_twist_12_host(p[0], p[1], p[2], pColorTwistParams);    // color_twist adjustment
+                    rpp_simd_store(rpp_store12_f32pln3_to_f32pkd3, dstPtrTemp_ps, p);    // simd stores
```

Comment:
We need to convert the pixels to fp16 and store in the SIMD code to avoid extra conversion at the end

Reply:
@rrawther I am unable to use the _mm256_loadu_ph() that directly loads half precision pixels since its under AVX512 instructions -> https://www.intel.com/content/www/us/en/docs/intrinsics-guide/index.html#text=_mm256_loadu_&ig_expand=7353,340,383,4325,4324